### ISSUE TITLE
Allow ignore option for one-var rule

### ISSUE THREAD

**Issue by applemonkey496** [eslint/eslint#14676]
**What rule do you want to change?**
The `one-var` rule.

**Does this change cause the rule to produce more or fewer warnings?**
It would depend on the configuration, but it would be an option to disable warnings in some cases.

**How will the change be implemented? (New option, new default behavior, etc.)?**
I propose an `ignore` option for the `var`/`const`/`let` rule. The default is `always` for all of these, and it would be useful to be able to lint just one of them. Example:

```js
/*eslint one-var: ['error', { 'const': 'never' }]*/

function foo() {
    const bar = 4
    const baz = 5
    
    // error
    let qux = 6
    let quux = 7
}
```

Here, I only specify that I never want multiple `const` declarations in one statement, but this should also create an error with the `let` declarations, because the default rule says that they should but put together into one statement. 

With an `ignore` option, this could be avoided by using:
```js
/*eslint one-var: ['error', { 'const': 'never', 'let': 'ignore', 'var': 'ignore' }]*/
```

Now, ESLint will disregard whether `let` or `var` declarations are together or separate, but make sure that `const` declarations are never done in one statement. This is just one example of how this would be useful. The same idea here could be applied to any combination of `let`, `const`, and `var`, for `never` and/or `consecutive`.

**Are you willing to submit a pull request to implement this change?**
I'm not really sure how the ESLint codebase is structured, so I probably won't be able to work on this myself.

**Comment by ljharb** [comment]
Why would you want them to be inconsistent?

(Note that the debugger experience is significantly worse when there's not always one var/let/const per variable)

**Comment by applemonkey496** [comment]
For me, I don't necessarily care whether `let` or `var` declarations are written as one statement or multiple statements, but for `const` I have a preference for separate declarations. I find it more useful to have `const` in front of each declaration so I can see that and remember not to accidentally reassign them.

Although not everyone would agree with my reasoning here, I don't think that this would require a ton of effort to implement anyways. It's mainly just a matter of an `if` block checking for the `ignore` option before raising errors.

**Comment by ljharb** [comment]
Given that, why not just force separate declarations for everything? (as all the common styleguides and eslint configs have mandated for many years)

**Comment by applemonkey496** [comment]
I suppose, yes 👍
